src/libaktualizr/uptane/uptane_test.cc

```diff
@@ -918,6 +1090,36 @@ TEST(Uptane, offlineIteration) {
   EXPECT_TRUE(sota_client->uptaneOfflineIteration(&targets_offline, nullptr));
   EXPECT_EQ(targets_online, targets_offline);
 }
+/*
+ Ignore updates for unrecognized ECUs.
+*/
+TEST(Uptane, DISABLED_IgnoreUnknownUpdate) {
+  // FIXME This test is disabled because it conflicts with some tests in uptane_vector_tests.cc
+  TemporaryDirectory temp_dir;
+  auto http = std::make_shared<HttpFake>(temp_dir.Path(), "hasupdates");
+  Config config("tests/config/basic.toml");
+  config.storage.path = temp_dir.Path();
+  config.uptane.director_server = http->tls_server + "director";
+  config.uptane.repo_server = http->tls_server + "repo";
+  config.pacman.type = PackageManager::kNone;
+  config.provision.primary_ecu_serial = "primary_ecu";
+  config.provision.primary_ecu_hardware_id = "primary_hw";
+  UptaneTestCommon::addDefaultSecondary(config, temp_dir, "secondary_ecu_serial", "secondary_hw");
+  config.postUpdateValues();
+
+  auto storage = INvStorage::newStorage(config.storage);
+  auto sota_client = SotaUptaneClient::newTestClient(config, storage, http);
+
+  EXPECT_NO_THROW(sota_client->initialize());
+  sota_client->AssembleManifest();
+
+  std::vector<Uptane::Target> targets;
+  EXPECT_TRUE(sota_client->uptaneIteration());
+  EXPECT_TRUE(sota_client->getNewTargets(&targets));
+
+  EXPECT_EQ(targets.size(), 1);
+  EXPECT_EQ(targets[0].filename(), "secondary_firmware.txt");
```

Comment:
We need to do just a bit more here to prove that nothing was actually updated and the targets were ignored.

Reply:
Also, there are two places that we need to verify that unknown targets are ignored: during the fetch and during the actual check for updates. Seems a bit redundant at a high level, but worth checking both, and since most of the work is already done here, it seems like it should be easy to check both here.

Reply:
Actually, we can probably cover the `Reject targets which do not match a known ECU` case here as well. Just try installing the bogus target and verify that it is rejected.

Reply:
@patriotyk Exactly! They are the same. Perhaps using the word "ignore" was confusing. But for now the correct thing to do is to reject and complain.

Reply:
Ok now I understand, it is the same but first one for installing second one for downloading

Reply:
Close. `Reject targets which do not match a known ECU` is part of `Install updates`, but `Ignore updates for unrecognized ECUs` is part of both `Fetch metadata from server` and `Check for updates`. Currently we have a lot of redundancy between those last two steps but we haven't yet come to a conclusion on what the right thing to with all that is. For now, we should cover all of those cases.

Reply:
I have added "Reject targets which do not match a known ECU" test case